packages/app/components/stackbox/FrequencyOptionsCard.tsx

```diff
@@ -0,0 +1,130 @@
+"use client";
+
+import { ChangeEvent, useEffect, useState } from "react";
+
+import { add } from "date-fns";
+import { twMerge } from "tailwind-merge";
+
+import { FREQUENCY_OPTIONS } from "@/models";
+import { BodyText, RadioButton, TextInput } from "@/ui";
+
+interface FrequencyOptionsCardProps {
+  frequency: FREQUENCY_OPTIONS;
+  setEndDate: (date: Date) => void;
+}
+
+const defaultFrequencyOptions = {
+  [FREQUENCY_OPTIONS.hour]: ["6", "12", "24"],
+  [FREQUENCY_OPTIONS.day]: ["7", "15", "30"],
+  [FREQUENCY_OPTIONS.week]: ["4", "8", "12"],
+  [FREQUENCY_OPTIONS.month]: ["2", "6", "12"],
+};
+const maxCustomFrequencies = {
+  [FREQUENCY_OPTIONS.hour]: 96,
+  [FREQUENCY_OPTIONS.day]: 365,
+  [FREQUENCY_OPTIONS.week]: 52,
+  [FREQUENCY_OPTIONS.month]: 24,
+};
+const postiveIntegerOnly = /^$|^[0-9]+$/;
+
+const getEndDateByDefaultFrequency = (
+  frequency: FREQUENCY_OPTIONS,
+  timeAmount: number
+) => {
+  switch (frequency) {
+    case FREQUENCY_OPTIONS.hour:
+      return add(new Date(), { hours: timeAmount });
+    case FREQUENCY_OPTIONS.day:
+      return add(new Date(), { days: timeAmount });
+    case FREQUENCY_OPTIONS.week:
+      return add(new Date(), { weeks: timeAmount });
+    case FREQUENCY_OPTIONS.month:
+      return add(new Date(), { months: timeAmount });
+    default:
+      throw new Error("Invalid frequency option");
```

Comment:
Let's not throw an error here, because we aren't catching it. Maybe a console.error is good enough in this case.